app/lib/frontend/backend.dart

```diff
@@ -77,9 +77,12 @@ class Backend {
   }
 
   /// Retrieves the names of all packages, ordered by name.
-  Stream<String> allPackageNames() {
+  Stream<String> allPackageNames({DateTime updatedSince}) {
     final query = db.query(models.Package);
-    return query.run().map((p) => (p as models.Package).name);
+    return query.run().where((p) {
+      return updatedSince == null ||
```

Comment:
why not put this into the query?

Reply:
Oh!...uh...I guess I could. Wasn't sure how...

Reply:
Done!